crates/typst-syntax/src/package.rs

```diff
@@ -216,6 +216,171 @@ impl PackageVersion {
     }
 }
 
+impl PackageVersion {
+    // the code in this impl block is based on https://github.com/dtolnay/semver/blob/1.0.23/src/eval.rs
+    // author: David Tolnay <https://github.com/dtolnay>
+    // licensed under Apache 2.0
```

Comment:
If the code is really close enough to the original, then we'd also need an entry in the NOTICE file. though to me it looks fairly different.

Reply:
yeah, I wanted to be on the safe side and at least give you the opportunity to judge for yourself. If you agree that it's fairly different (algorithmically it's similar, but the code itself in many parts isn't) then I'll remove that note.

Reply:
yeah, afaik copyright doesn't cover algorithms, just code, so I think we don't need it. that aside, the algorithm is also pretty obvious.